### PR TITLE
ops agent self metrics

### PR DESCRIPTION
**Pull request by franciscovalentecastro**
Draft

### CODE REVIEW

File: cmd/google_cloud_ops_agent_engine/main.go [hunk 1]
```diff
@@ -59,5 +61,19 @@ func run() error {
 	if err != nil {
 		return err
 	}
-	return confgenerator.GenerateFilesFromConfig(&uc, *service, *logsDir, *stateDir, *outDir)
+	err = confgenerator.GenerateFilesFromConfig(&uc, *service, *logsDir, *stateDir, *outDir)
+	if err != nil {
+		return fmt.Errorf("The agent config file is not valid. Detailed error: %s", err)
```
Comment (sophieyfang) [inline cmd/google_cloud_ops_agent_engine/main.go:66]: return err

File: internal/monitoring_api/monitoring_api.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	timestamp "github.com/golang/protobuf/ptypes/timestamp"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoredres "google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+)
+
+func constructTimeSeriesRequest(ctx context.Context, eR confgenerator.EnabledReceivers) (*monitoringpb.CreateTimeSeriesRequest, error) {
+	creds, err := google.FindDefaultCredentials(ctx)
+	if err != nil {
```
Comment (franciscovalentecastro) [inline internal/monitoring_api/monitoring_api.go:38]: - Rename google to oauth2
- Rename metadata to gce_metadata

File: cmd/google_cloud_ops_agent_engine/main.go [hunk 3]
```diff
@@ -59,5 +61,22 @@ func run() error {
...
+	if *service ==  "" { 
+		eR, err := confgenerator.GetEnabledReceivers(uc)
+		if err != nil {
+			return err
+		}
+		log.Println("Enabled Receivers", eR)
+		err = monitoring_api.SendMetricEveryInterval(eR, 60)
+		if err != nil {
+			return err
+		}
```
Comment (avilevy18) [inline cmd/google_cloud_ops_agent_engine/main.go:79]: Move this into a method where we can register multiple metrics at different intervals

File: internal/monitoring_api/monitoring_api.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+	err = c.CreateTimeSeries(ctx, req)
+	if err != nil {
+		return fmt.Errorf("could not write time series value, %v ", err)
+	}
+
+	return nil
+}
+
+func SendMetricEveryInterval(eR confgenerator.EnabledReceivers, interval int) error {
```
Comment (avilevy18) [inline internal/monitoring_api/monitoring_api.go:128]: Maybe make a generic Metric struct as instead of eR

**Pull request closed without merging**